contracts/router/quoter/PoolQuoterV1.sol

```diff
@@ -0,0 +1,396 @@
+    /// - If one of the pool tokens is WETH, ETH_ADDRESS is also considered as a pool token.
+    /// Note: this is not supposed to be used with LinkedPool contracts, as a single token can appear
+    /// multiple times in the LinkedPool's token tree.
+    function _getTokenIndexes(
+        address pool,
+        address tokenIn,
+        address tokenOut
+    ) internal view returns (uint8 indexIn, uint8 indexOut) {
+        uint256 numTokens = _numTokens(pool);
+        // Assign NOT_FOUND to both indexes by default. This value will be overwritten if the token is found.
+        indexIn = NOT_FOUND;
+        indexOut = NOT_FOUND;
+        unchecked {
+            // unchecked: numTokens <= 255 => ++t never overflows uint8
+            for (uint8 t = 0; t < numTokens; ++t) {
+                address poolToken = IDefaultExtendedPool(pool).getToken(t);
+                if (_poolToken(tokenIn) == poolToken) indexIn = t;
+                if (_poolToken(tokenOut) == poolToken) indexOut = t;
+            }
+        }
+    }
+
+    /// @dev Checks if a pool conforms to IPausable interface, and if so, returns its paused state.
+    /// Returns false if pool does not conform to IPausable interface.
+    function _isPoolPaused(address pool) internal view returns (bool) {
+        // We issue a static call in case the pool does not conform to IPausable interface.
+        (bool success, bytes memory returnData) = pool.staticcall(abi.encodeWithSelector(IPausable.paused.selector));
+        // Pool is paused if the call was successful and returned true.
+        // We check the return data length to ensure abi.decode won't revert.
+        return success && returnData.length == 32 && abi.decode(returnData, (bool));
+    }
+
+    // ════════════════════════════════════════ POOL TOKEN -> TOKEN QUOTES ═════════════════════════════════════════════
+
+    /// @dev Checks whether `tokenIn -> tokenOut` is possible via the given Default Pool, given the
+    /// `actionMask` of available actions for the token.
+    /// Note: only checks DefaultPool-related actions: Swap/AddLiquidity/RemoveLiquidity.
+    function _isConnectedViaDefaultPool(
+        uint256 actionMask,
+        address pool,
+        address tokenIn,
+        address tokenOut
+    ) internal view returns (bool) {
+        // We don't check for paused pools here, as we only need to know if a connection exists.
+        (uint8 indexIn, uint8 indexOut) = _getTokenIndexes(pool, tokenIn, tokenOut);
+        // Check if Swap (tokenIn -> tokenOut) could fulfill tokenIn -> tokenOut request.
+        if (Action.Swap.isIncluded(actionMask) && indexIn != NOT_FOUND && indexOut != NOT_FOUND) {
+            return true;
+        }
+        address lpToken = _lpToken(pool);
+        // Check if AddLiquidity (tokenIn -> lpToken) could fulfill tokenIn -> tokenOut request.
+        if (Action.AddLiquidity.isIncluded(actionMask) && indexIn != NOT_FOUND && tokenOut == lpToken) {
```

Comment:
Might want to also check that `indexOut == NOT_FOUND` for `addLiquidity`. Similarly, `indexIn == NOT_FOUND` for `removeLiquidity`

Reply:
This check could be omitted, as `tokenOut == lpToken` inevitably leads to `tokenOut` not being in the pool, and therefore `NOT_FOUND` value.